### ISSUE TITLE
[BUG] kaggle platform and GPU crashes

### ISSUE THREAD

**Issue by mglowacki100** [autogluon/autogluon#3365]
1. Kaggle notebook crashes during training when gpu is enabled, with following error (or without it):
```
Fitting 31 L1 models ...
Fitting model: KNeighborsUnif_BAG_L1 ... Training model for up to 3577.57s of the 3577.53s of remaining time.
	-0.9982	 = Validation score   (-log_loss)
	1.02s	 = Training   runtime
	0.33s	 = Validation runtime
Fitting model: KNeighborsDist_BAG_L1 ... Training model for up to 3575.61s of the 3575.57s of remaining time.
	-0.9845	 = Validation score   (-log_loss)
	0.34s	 = Training   runtime
	0.34s	 = Validation runtime
Fitting model: LightGBMXT_BAG_L1 ... Training model for up to 3574.32s of the 3574.28s of remaining time.
	Fitting 8 child models (S1F1 - S1F8) | Fitting with ParallelLocalFoldFittingStrategy
2023-06-29 07:08:26,189	ERROR services.py:1169 -- Failed to start the dashboard , return code -11
2023-06-29 07:08:26,191	ERROR services.py:1194 -- Error should be written to 'dashboard.log' or 'dashboard.err'. We are printing the last 20 lines for you. See 'https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure' to find where the log file is.
2023-06-29 07:08:26,195	ERROR services.py:1238 -- 
The last 20 lines of /tmp/ray/session_2023-06-29_07-08-24_401222_28/logs/dashboard.log (it contains the error message from the dashboard): 
2023-06-29 07:08:26,125	INFO head.py:239 -- Starting dashboard metrics server on port 44227
```
2. Here is complete notebook: https://www.kaggle.com/code/pareto80/biol-gpu/notebook **version 2**, version 1 was ran without accelerator and it works fine. Competition page: https://www.kaggle.com/competitions/bioresponse/overview
3. Latest environment on kaggle is python 3.10.10 so this is not problem with python version 
4. I've also tried without lightgbm gpu installation and other combinations.

**Comment by trhgquan** [comment]
Hi, so I just ran into this problem while trying to run AutoGluon on Kaggle with P100 and T4 GPU. It's the same error, the same error message, and disabling the accelerator would eventually make things work. @mglowacki100 did you find a way to resolve it?

**Comment by mglowacki100** [comment]
@trhgquan I didn't look into this, but I think it is caused by ray: https://discuss.ray.io/t/ray-crashed-jupyter-notebook/12262 so I'd try to upgrade/downgrade it

**Comment by trhgquan** [comment]
Hi, so I just resolved the problem. Since Kaggle's default environment has `ray==2.9.0` and AutoGluon requires `ray>=2.6.3, ray<=2.7.0`, `pip` automatically uninstall `ray==2.9.0` to replace with `ray==2.6.3`. However, this caused the bug.

I tried reinstalling `ray==2.9.0`, then again, installing `ray==2.6.3` and the problem was solved unexpectedly. Guess there are some package conflicts between the Kaggle environment and AutoGluon's requirements itself, though it needs further experiments to point out.

Anyway, huge appreciate your idea, @mglowacki10!
